docs/pages/router/reference/authentication.mdx

```diff
@@ -5,115 +5,352 @@ description: How to protect routes with Expo Router.
+
+<FileTree
+  files={[
+    ['app/_layout.js', 'Declares global session context'],
+    'app/(app)/_layout.js',
+    ['app/(app)/sign-in.js', <span>Modally presented over the root</span>],
+    ['app/(app)/(root)/_layout.js', <span>Protects child routes</span>],
+    [
+      'app/(app)/(root)/index.js',
+      <span>
+        Requires authorization <Lock01Icon className="inline mb-1" />
+      </span>,
+    ],
+  ]}
+/>
+
+```jsx app/(app)/_layout.js
+import { Stack } from 'expo-router';
+
+export const unstable_settings = {
+  initialRouteName: '(root)',
+};
+
+export default function AppLayout() {
+  return (
+    <Stack>
+      <Stack name="(root)" />
+      <Stack
+        name="sign-in"
+        options={{
+          presentation: 'modal',
+        }}
+      />
+    </Stack>
+  );
+}
+```
+
+## Navigating without navigation
+
+If you encounter the following error:
+
+```
+Error: Attempted to navigate before mounting the Root Layout component. Ensure the Root Layout component is rendering a Slot, or other navigator on the first render.
+```
+
+This can occur when the app attempts to perform navigation without a navigator mounted in the [root layout](/router/advanced/root-layout). You can fix this by adding a group and moving conditional logic up a level.
+
+### Before
+
+<FileTree files={['app/_layout.js', 'app/about.js']} />
+
+```js app/_layout.js
+export default function RootLayout() {
+  React.useEffect(() => {
+    // This navigation event will trigger the error above.
+    router.push('/about');
+  }, []);
+
+  // Avoid deferring the root layout, otherwise the navigation event will be triggered before the layout is mounted.
+  if (isLoading) {
+    return <Text>Loading...</Text>;
+  }
+
+  return <Slot />;
+}
+```
+
+### After
+
+<FileTree
+  files={[
+    ['app/_layout.js', ''],
+    ['app/(app)/_layout.js', 'Move conditional logic up a level'],
+    'app/(app)/about.js',
+  ]}
+/>
+
+```js app/_layout.js
+export default function RootLayout() {
+  return <Slot />;
+}
+```
+
+```js app/(app)/_layout.js
+export default function RootLayout() {
+  React.useEffect(() => {
+    // This navigation event will trigger the error above.
+    router.push('/about');
+  }, []);
+
+  // Avoid deferring the root layout, otherwise the navigation event will be triggered before the layout is mounted.
+  if (isLoading) {
+    return <Text>Loading...</Text>;
+  }
+
+  return <Slot />;
+}
+```
+
+## Middleware
 
-{/* TODO: Guide on using redirects and per-screen behavior */}
+Traditionally, websites may leverage some form of server-side redirection to protect routes. As of Expo Router v2, Expo Router on web currently only supports build-time static generation and has no support for custom middleware or serving. This can be added in the future to provide a more optimal web experience, in the meantime, authentication can be implemented by using client-side redirects and a loading state.
```

Comment:
```suggestion
Traditionally, websites may leverage some form of server-side redirection to protect routes. As of Expo Router v2, Expo Router on web currently only supports build-time static generation and has no support for custom middleware or serving. This can be added in the future to provide a more optimal web experience and, in the meantime, authentication can be implemented by using client-side redirects and a loading state.
```